sysml.library/Kernel Library/SpatialFrames.kerml

```diff
@@ -0,0 +1,160 @@
+/**
+ * This package models spatial frames of reference for quantifying the position of points 
+ * in a three-dimensional space. 
+ */
+package SpatialFrames {
+	private import Clocks::*;
+	private import ScalarValues::NumericalValue;
+	private import VectorValues::ThreeVectorValue;
+	private import VectorValues::CartesianThreeVectorValue;
+	private import VectorFunctions::isZeroVector;
+	private import Objects::Body;
+	private import Objects::Point;
+	private import ControlFunctions::forAll;
+	private import SequenceFunctions::includes;
+	
+	/**
+	 * defaultFrame is a fixed SpatialFrame used as a universal default.
+	 */
+	readonly feature defaultFrame : SpatialFrame[1];
+	
+	/**
+	 * A SpatialFrame is a three-dimensional Body that provides a spatial extent that 
+	 * acts as a frame of reference for defining the phyiscal position and displacement 
+	 * vectors of Points over time.
+	 */
+	abstract struct SpatialFrame specializes Body;
```

Comment:
I find it slightly non-intuitive to make SpatialFrame which I would consider intangible (a mental abstraction) to specialize Body which I would intuitively consider tangible. I know Body is defined to be a 3D spatial extent (finite region), but then the label "body" is perhaps not appropriate. In geometry often "Solid" is used to denote a finite 3D extent (as indeed STEP and my current Shapes.sysml does), although that is misleading in a different way as it may be a volume containing a non-solid state of aggregation: liquid, gas, plasma, ...
Can SpatialFrame be a [1..1] feature of Body, rather than being a Body?

Reply:
Could have a different name for Body, but then Object has similar dissonance, bc it's just an Occurrence that (informally stated in the spec) occupies a connected region of space time.  This covers things you can pick up ("tanglible") and things you can't, eg, a piece of dark matter.

Reply:
A SpatialFrame is intended to be an Object (as Clock is) with a three-dimensional spatial extent. Naming aside, this is just what a Body is, tangible or intangible (as Conrad notes) – and there probably is no perfect naming. So I think it makes sense for SpatialFrame to be a Body.